File: 02-HibernateJPA/src/main/java/hibernate/Departamento.java
package hibernate;

import javax.persistence.*;
import java.util.Collection;
import java.util.Set;

@Entity
@NamedQuery(name = "Departamento.finAll", query = "SELECT d FROM Departamento d")
public class Departamento {
    private long id;
    private String nombre;

    private Collection<Empleado> empleados;

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name = "id")
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @Basic
    @Column(name = "nombre")
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Departamento that = (Departamento) o;

        if (id != that.id) return false;
        if (nombre != null ? !nombre.equals(that.nombre) : that.nombre != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = (int) (id ^ (id >>> 32));
        result = 31 * result + (nombre != null ? nombre.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Departamento{" +
                "id=" + id +
                ", nombre='" + nombre + '\'' +
                '}';
    }

    @OneToMany(mappedBy = "departamento")
    public Collection<Empleado> getEmpleados() {
        return empleados;
    }

    public void setEmpleados(Collection<Empleado> empleados) {
        this.empleados = empleados;
    }
}

File: 04-KotlinJPA/src/main/kotlin/hibernate/Departamento.kt
package hibernate

import javax.persistence.*


@Entity
@NamedQuery(name = "Departamento.finAll", query = "SELECT d FROM Departamento d")
class Departamento {
    @get:Column(name = "id")
    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
    @get:Id
    var id: Long = 0

    @get:Column(name = "nombre")
    @get:Basic
    var nombre: String? = null

    @get:OneToMany(mappedBy = "departamento")
    var empleados: Collection<Empleado>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Departamento

        if (id != other.id) return false
        if (nombre != other.nombre) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + (nombre?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "Departamento(id=$id, nombre=$nombre, empleados=$empleados)"
    }


}
